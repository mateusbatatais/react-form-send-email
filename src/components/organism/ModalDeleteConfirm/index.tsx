import React, { useState } from "react";
import { Modal, Button } from "react-bootstrap";
import api from "../../../services/api";

interface Props {
  id: number;
  onDelete: () => void;
}

function ModalDeleteConfirm({ id, onDelete }: Props) {
  const [show, setShow] = useState<boolean>(false);
  const handleClose = () => setShow(false);
  const handleShow = () => setShow(true);

  const deleteItem = () => {
    api.delete(`/Contact/${id}`).then((response) => {
      handleClose();
      onDelete();
    });
  };

  return (
    <>
      <Modal show={show} onHide={handleClose}>
        <Modal.Header closeButton>
          <Modal.Title>Confirma exclusão?</Modal.Title>
        </Modal.Header>
        <Modal.Body>
          Cuidado, essa ação não tem volta! Podemos excluir?
        </Modal.Body>
        <Modal.Footer>
          <Button variant="secondary" onClick={handleClose}>
            Voltar
          </Button>
          <Button variant="danger" onClick={deleteItem}>
            Excluir
          </Button>
        </Modal.Footer>
      </Modal>
      <Button
        variant="danger"
        className="me-1 mb-1"
        size="sm"
        name="deletar"
        onClick={handleShow}
      >
        Excluir
      </Button>
    </>
  );
}

export default ModalDeleteConfirm;
