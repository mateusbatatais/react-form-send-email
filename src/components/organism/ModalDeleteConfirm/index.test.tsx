import React from "react";
import { render } from "@testing-library/react";
import ModalDeleteConfirm from "./index";

describe("testes modal delete", () => {
  it("snapshot do botÃ£o", () => {
    const { container } = render(
      <ModalDeleteConfirm id={1} onDelete={() => ""} />
    );
    expect(container).toMatchSnapshot();
  });

  it("testa botÃ£o abrir", () => {});
});
