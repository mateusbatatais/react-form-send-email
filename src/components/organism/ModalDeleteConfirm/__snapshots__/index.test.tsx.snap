// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testes modal delete snapshot do bot√£o 1`] = `
<div>
  <button
    class="me-1 mb-1 btn btn-danger btn-sm"
    type="button"
  >
    Excluir
  </button>
</div>
`;
